evm-ds/src/evm_server_run.rs

```diff
@@ -225,8 +228,9 @@ fn build_exit_result(
                         modify.set_reset_storage(reset_storage);
                         let storage_proto = storage
                             .into_iter()
-                            .map(|(k, v)| backend.encode_storage(k, v).into())
+                            .map(|(k, v)| { continuations.lock().unwrap().update_states(address, k, v); backend.encode_storage(k, v).into()})
```

Comment:
applying state changes to other pending continuations - seems a little hacky to do it this way